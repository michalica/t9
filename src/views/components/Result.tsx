import React from 'react';
import styled from 'styled-components';
import MoreButton from "./MoreButton";

const List = styled.ul`
    list-style: none
    padding-left: 0px
`

const ListItem = styled.li`
    text-decoration: none
    margin-bottom: 15px
    font-size: 25px
    font-weight: bold
`


const Result = (props: {
    letters: Array<string>,
    handleShowMoreButtonClick: Function,
    isLoading: boolean,
}) => {
    if (!props.letters || props.letters.length == 0) {
        return <div></div>;
    }
    const items = props.letters.map((item, key) => {
        return <ListItem key={key}> {item} </ListItem>
    });
    return (
        <div>
            <List>
                {
                    items
                }
            </List>
            {
                props.letters.length >= 10 &&
                <MoreButton
                    text="+"
                    handleClick={props.handleShowMoreButtonClick}
                    isLoading={props.isLoading}
                />
            }

        </div>
    );
}


export default Result;
