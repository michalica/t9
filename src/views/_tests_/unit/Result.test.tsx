import React from 'react';
import Result from '../../components/Result';
import renderer from 'react-test-renderer';


import chai from 'chai';
import chaiEnzyme from 'chai-enzyme';

chai.use(chaiEnzyme());

const handler = jest.fn();

const testNames = [
    'renders without crashing without MoreButton',
    'renders without crashing with MoreButton',
    'renders without crashing empty'
];

const props = [
        {
            letters: ['abc', 'fds'],
            isLoading: true,
            handleShowMoreButtonClick: handler,
        },
        {
            letters: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11'],

            isLoading: true,
            handleShowMoreButtonClick: handler,
        },
        {
            letters: ['1'],
            isLoading: true,
            handleShowMoreButtonClick: handler,
        }
    ];

props.map((currentProps, key) => {
    testResult(currentProps, testNames[key]);

});



function testResult(currentProps:{
    letters: Array<string>,
    isLoading: boolean,
    handleShowMoreButtonClick: Function},
                    name:string) {
    it(name, () => {
        const tree = renderer
            .create(<Result {...currentProps} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
}
