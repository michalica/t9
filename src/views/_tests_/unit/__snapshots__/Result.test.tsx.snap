// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing empty 1`] = `
<div>
  <ul
    className="sc-bdVaJa loehwO"
  >
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      1
       
    </li>
  </ul>
</div>
`;

exports[`renders without crashing with MoreButton 1`] = `
<div>
  <ul
    className="sc-bdVaJa loehwO"
  >
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      1
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      2
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      3
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      4
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      5
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      6
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      7
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      8
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      9
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      10
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      11
       
    </li>
  </ul>
  <div>
    <button
      className="ui loading button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;

exports[`renders without crashing without MoreButton 1`] = `
<div>
  <ul
    className="sc-bdVaJa loehwO"
  >
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      abc
       
    </li>
    <li
      className="sc-bwzfXH erLWkn"
    >
       
      fds
       
    </li>
  </ul>
</div>
`;
